### PR TITLE
Add tenant admin access

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/9248

### CODE REVIEW

File: helm/rbac-operator/templates/rbac-tenant-admin.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  - cluster.x-k8s.io
+  - infrastructure.giantswarm.io
+  resources:
+  - appcatalogs
+  - apps
+  - clusters
+  - machinedeployments
+  - awsclusters
+  - awsmachinedeployments
+  verbs:
```
Comment (MarcelMue) [inline helm/rbac-operator/templates/rbac-tenant-admin.yaml]: Maybe add patch as well?

File: flag/service/namespaceauth/namespaceauth.go [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package namespaceauth
 
 type NamespaceAuth struct {
```
Comment (MarcelMue) [inline flag/service/namespaceauth/namespaceauth.go]: Alphabetical order would please me.

File: flag/service/namespaceauth/namespaceauth.go [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
-	ViewAllTargetGroup string
+	ViewAllTargetGroup     string
+	TenantAdminTargetGroup string
```
Comment (MarcelMue) [inline flag/service/namespaceauth/namespaceauth.go]: So tenant admin is a customer admin with access to the CP k8s, right?

File: helm/rbac-operator/templates/rbac-tenant-admin.yaml (hunk 1 above)
Comment (corest) [reply helm/rbac-operator/templates/rbac-tenant-admin.yaml]: uff, mised that

File: flag/service/namespaceauth/namespaceauth.go (hunk 3 above)
Comment (corest) [reply flag/service/namespaceauth/namespaceauth.go]: Right. It has access to create app/appcatalogs and cluster/machindeployments CRs

File: helm/rbac-operator/templates/rbac-tenant-admin.yaml (hunk 1 above)
Comment (kopiczko) [reply helm/rbac-operator/templates/rbac-tenant-admin.yaml]: Can't we replace that with `"*"`?

File: service/controller/resource/namespaceauth/create.go [hunk 4]
```diff
@@ -21,51 +22,60 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	viewAccessWords := []string{"get", "list", "watch"}
-	viewAllRole, err := newViewAllRole(resources, viewAccessWords)
-	if err != nil {
-		return microerror.Mask(err)
+	viewAllRole.targetGroup = r.namespaceAuth.ViewAllTargetGroup
+	tenantAdminRole.targetGroup = r.namespaceAuth.TenantAdminTargetGroup
```
Comment (kopiczko) [inline service/controller/resource/namespaceauth/create.go]: Do we have to have globally defined variables and mutate them in some place? Can't we define them here instead?

File: helm/rbac-operator/templates/rbac-tenant-admin.yaml (hunk 1 above)
Comment (corest) [reply helm/rbac-operator/templates/rbac-tenant-admin.yaml]: Yes, we can. Just wanted to be explicit here

File: service/controller/resource/namespaceauth/create.go (hunk 4 above)
Comment (corest) [reply service/controller/resource/namespaceauth/create.go]: It is also in `delete.go` so I didn't want to duplicate it

File: helm/rbac-operator/templates/rbac-tenant-admin.yaml (hunk 1 above)
Comment (kopiczko) [reply helm/rbac-operator/templates/rbac-tenant-admin.yaml]: In this case this explicitly states you can do anything. But I won't argue.

File: service/controller/resource/namespaceauth/create.go (hunk 4 above)
Comment (kopiczko) [reply service/controller/resource/namespaceauth/create.go]: Replacing those vars with `newViewAllRole` funcs would work for. Otherwise I feel insecure :D

Comment (corest) [reply service/controller/resource/namespaceauth/create.go]: Ok
I've moved vars into function as we actually don't need all of them in delete function

**Comment by corest** [comment]
PTAL

**Pull request merged**